### PR TITLE
Add sort by profit functionality to list command

### PR DESCRIPTION
**Pull request by nur-haziq**
Add ability for user to specify the `-p` flag in the already existing `list` command . This returns a sorted list, consisting only of retail items (since only sellable items can generate revenue), in descending order of profits.

Towards #128 

@PureUsagi Kindly assist in checking if I have implemented the code correctly, as it relies heavily on your already existing list command.

-------------------------------------------------------------------------------
Here's a sample dataset that you can use for items.txt:

PerishableRetailItem|Apples|Red Delicious|250|0.4|250|50|10-06-2024|1.2|125|
OperationalItem|Sprinkler|Oscillating Sprinkler|15|35.0|15|3| | | |
PerishableOperationalItem|Insecticide|Organic Neem Oil|90|12.0|90|15|01-08-2024| | |
RetailItem|Pruning Shears|Stainless Steel|35|10.0|35|7| |29.99|18|
PerishableRetailItem|Oranges|Navel|300|0.45|300|60|15-07-2024|1.5|150|
OperationalItem|Lawn Mower|Electric Lawn Mower|10|150.0|10|2| | | |
PerishableOperationalItem|Fertilizer|Liquid Fish Emulsion|75|20.0|75|10|20-09-2024| | |
RetailItem|Garden Fork|Ergonomic Handle|25|9.0|25|5| |27.99|12|
PerishableRetailItem|Carrots|Organic|200|0.25|200|40|30-05-2024|0.75|100|
OperationalItem|Wheelbarrow|Steel Wheelbarrow|8|60.0|8|2| | | |
PerishableOperationalItem|Weed Killer|Non-toxic Formula|85|14.0|85|10|15-10-2024| | |
RetailItem|Hedge Trimmer|Cordless|20|80.0|20|4| |199.99|10|
PerishableRetailItem|Strawberries|Fresh Picked|180|0.7|180|30|05-06-2024|1.8|90|
OperationalItem|Seed Spreader|Broadcast Spreader|12|45.0|12|3| | | |
PerishableOperationalItem|Mulch|Cedar Bark|100|5.0|100|20|01-12-2024| | |
RetailItem|Gardening Gloves|Leather|60|3.5|60|10| |12.99|30|
PerishableRetailItem|Peaches|Yellow Flesh|220|0.55|220|45|20-08-2024|1.25|110|
OperationalItem|Garden Hose Reel|Wall Mounted|10|25.0|10|2| | | |
PerishableOperationalItem|Plant Food|All-purpose|70|8.0|70|10|28-02-2025| | |
RetailItem|Watering Can|Galvanized Steel|40|7.0|40|8| |19.99|20|

### CODE REVIEW

**Review by YHWong20 — APPROVED** [review]
Looks good to me. I've done a quick test of the `list -p` command as well. Seems to be working fine.

**Review by nkotaa — APPROVED** [review]
Points to be addressed are non-critical. Thanks for documenting your methods with sufficient Javadocs.

**Review by PureUsagi — COMMENTED** [review]
The implementation of sort looks good! Just a few minor comments, that can be ignored as they do not affect the correctness of the sort.

File: src/main/java/seedu/binbash/comparators/ItemComparatorByProfit.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+     * @param item2 The second object to be compared.
+     * @return A negative integer if the first item's profit is greater than the second item's profit,
+     *         positive integer if the first item's profit is less than or equals to the second item's profit,
+     */
+    public int compare(Item item1, Item item2) {
+        retailItem1 = (RetailItem) item1 ;
+        retailItem2 = (RetailItem) item2;
+
+        if (retailItem1.getItemProfit() > retailItem2.getItemProfit()) {
+            return -1;
```
[the remark this answers is not in the document]
Comment (nur-haziq) [reply src/main/java/seedu/binbash/comparators/ItemComparatorByProfit.java:30]: Ah, overlooked that. Thanks!

**Pull request merged**